core/admin/mailu/ui/views/domains.py

```diff
@@ -70,6 +70,22 @@ def domain_details(domain_name):
     domain = models.Domain.query.get(domain_name) or flask.abort(404)
     return flask.render_template('domain/details.html', domain=domain)
 
+@ui.route('/domain/details/<domain_name>/zonefile', methods=['GET'])
+@access.domain_admin(models.Domain, 'domain_name')
+def domain_download_zonefile(domain_name):
+    domain = models.Domain.query.get(domain_name) or flask.abort(404)
+    final = domain.dns_mx+"\n"
+    final = final + domain.dns_spf+"\n"
+    if domain.dkim_publickey:
+        final = final + domain.dkim_publickey+"\n"
+        final = final + domain.dns_dkim+"\n"
+        final = final + domain.dns_dmarc+"\n"
+    if domain.dns_tlsa:
+        final = final + domain.dns_tlsa
+    for i in domain.dns_autoconfig:
+        final = final + i+"\n"
+    return flask.Response(final,content_type="text/plain",headers={'Content-disposition': 'attachment; filename='+domain.name+'-zonefile.txt'})
```

Comment:
LGTM.

Can you create the zone in a more pythonic way like this?

```suggestion
    res = [domain.dns_mx, domain.dns_spf]
    if domain.dkim_publickey:    
        res.extend([domain.dkim_publickey, domain.dns_dkim, domain.dns_dmarc])
    if domain.dns_tlsa:
        res.append(domain.dns_tlsa)
    res.extend(domain.dns_autoconfig)
    res.append("")
    return flask.Response("\n".join(res), content_type="text/plain", headers={"Content-disposition": f"attachment; filename={domain.name}-zonefile.txt"})
```

Reply:
And to handle #2618 we should change the property `dkim_publickey` to return a non-splitted record.
The record then needs to be split for the zonefile.
I think I'll update your PR to reflect this.

Reply:
> And to handle #2618 we should change the property `dkim_publickey` to return a non-splitted record. The record then needs to be split for the zonefile. I think I'll update your PR to reflect this.

Go ahead, I just really wanted this to exist, as I noticed the admin panel already spit out data in this format.